### DISCUSSION TITLE
localhost refused to connect.

### DISCUSSION THREAD

**Discussion by chetan7330** [community/community#113703] [Programming Help]
### Body

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:80:19)
    at Object.createHash (node:crypto:139:10)
    at module.exports (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/NormalModule.js:417:16)
    at handleParseError (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/NormalModule.js:471:10)
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/NormalModule.js:503:5
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/NormalModule.js:358:12
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at iterateNormalLoaders (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/react-scripts/scripts/start.js:19
  throw err;
  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:80:19)
    at Object.createHash (node:crypto:139:10)
    at module.exports (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/util/createHash.js:135:53)
    at NormalModule._initBuildHash (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/NormalModule.js:417:16)
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/NormalModule.js:452:10
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/webpack/lib/NormalModule.js:323:13
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/Users/chetankrishna/Desktop/Webathon/frontend/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at /Users/chetankrishna/Desktop/Webathon/frontend/node_modules/babel-loader/lib/index.js:59:103 {
  opensslErrorStack: [
    'error:03000086:digital envelope routines::initialization error',
    'error:0308010C:digital envelope routines::unsupported'
  ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v21.7.1

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ettaboyle** [comment]
Hi @chetan7330, we're happy you're here! 

You are more likely to get a useful response if you are explicit about what your project entails, giving a few more details might help someone give you a nudge in the right direction.

**Comment by developer-sumit** [comment]
The specific error `0308010C:digital envelope routines::unsupported` suggests that there might be an unsupported cryptographic operation being attempted. This can sometimes occur due to incompatibilities or issues with the version of Node.js being used, as well as potential issues with dependencies or configuration.

You can try the solutions written below and don't forget to mark this as answered if you like this:

1. **Check Node.js Version Compatibility**: Ensure that the version of Node.js you are using is compatible with the dependencies of your project. The error you provided indicates you're using Node.js version `21.7.1`. While Node.js versions in the 21.x range might be experimental or unstable at this time, it's worth considering using a more stable version, such as a LTS (Long Term Support) release.

2. **Update Dependencies**: Make sure all your project dependencies, including webpack and related modules, are up to date. Sometimes, compatibility issues can arise due to outdated dependencies.

3. **Check webpack Configuration**: Review your webpack configuration to see if there are any settings related to cryptographic operations or hashing algorithms that might be causing the issue. You may need to adjust these configurations or dependencies to resolve the problem.

4. **Check for Environment Issues**: Ensure that there are no environmental issues specific to your setup, such as conflicting installations or misconfigurations. Try running your project on a different machine or in a clean environment to see if the issue persists.

5. **Search for Known Issues**: Search online forums, GitHub issues, or Stack Overflow for similar errors. Sometimes, other developers have encountered the same problem and may have found solutions or workarounds.

6. **Consider Rolling Back Node.js Version**: If the issue persists and you suspect it's related to the Node.js version, consider rolling back to a stable version (e.g., the latest LTS release) to see if that resolves the problem.

7. **Debugging**: If none of the above steps resolve the issue, you may need to dig deeper into the webpack and Node.js internals to debug the problem further. This might involve logging, stepping through code, or using debugging tools to identify the root cause.
